blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/Containers.java

```diff
@@ -15,7 +15,133 @@
 
     public abstract static class AbstractContainer<T extends Resource> extends Container<T> {
     }
+    
+    /**
+     * Combines multiple {@link Iterable} to treat them as a single one
+     *
+     * @param <T> iterable type
+     */
+    public static class CombinedIterable<T> implements Iterable<T> {
```

Comment:
@kzantow This change is for very specific use case, that is activity API, where different objects (run and queue) are returned in single iterator and hence need to combine run and queue iterators in to single with intent to enforce pagination parameters across different iterators.

We already have abstraction to enforce pagination criteria over an iterator. I am not convinced this set of changes are needed as such common abstractions would make sense if we have such general use cases.

I have done an alternate implementation that pretty much does same. There is Pageables.combine() that combines iterator with given pagination criteria. There is also ActivityContainer, that uses Pageables.combine(). Its matter of taste but instead of ActivityContainer an anonymous inner class of Container implementation would have worked too.

Another thing, I don't think we can support /activities/:id call reliably so we should not support that. No matter whether you use queue id or run id. So ActivityContainer.get() in my impl always returns null. Instead of /activities/:id, UI should really be using link _links.self.href on each item in activities to fetch individual items.

See my alternate proposal at: https://github.com/jenkinsci/blueocean-plugin/commit/e631fa59144e731925aceecda1e929792921593d.